src/webots/nodes/utils/WbSupervisorUtilities.cpp

```diff
@@ -294,10 +294,12 @@ void WbSupervisorUtilities::initControllerRequests() {
   mFoundNodeParentUniqueId = -1;
   mFoundNodeIsProto = false;
   mFoundNodeIsProtoInternal = false;
+  mNodeFieldCount = -1;
   mFoundFieldId = -2;
```

Comment:
Not directly related to this PR, but this seems to be a bit cryptic, enum or similar would be helpful

Reply:
To me this is a normal behavior for variables meant to contain positive numbers like sizes or count:
`-1` means that it is not set, values `> 0` represent the found value, it is like NULL for pointers.
In any case, I don't see how you could use an enumeration to achieve the same result.
Note that any other solution would probably require additional variables.

Reply:
I was thinking something like this:
```cpp
enum { NOT_REQUESTED = -2, NOT_FOUND = -1 };

// ...

mFoundFieldId = NOT_REQUESTED;
```

Reply:
I'm not sure that mixing enum with int values is good.
With enumeration I would expect that all the possible values are listed in the enumeration and this won't be the case here.

If you want to add something like this, then I think you should not use an enumeration but simply define separate global variables for `NOT_REQUESTED` and `NOT_FOUND`.
However, this doesn't not only affect the two variables used in this PR but many others. So, if we want to improve this, it has to be done in a separate PR.

Reply:
> If you want to add something like this, then I think you should not use an enumeration but simply define separate global variables for `NOT_REQUESTED` and `NOT_FOUND`.

I agree, macros are probably a better idea.

> However, this doesn't not only affect the two variables used in this PR but many others. So, if we want to improve this, it has to be done in a separate PR.

👍